### DISCUSSION TITLE
Building a Full-Stack Turbo Monorepo with Expo, Clerk, Drizzle, Supabase, and tRPC

### DISCUSSION THREAD

**Discussion by Mo-Xiyad** [notJust-dev/notjustdev#271] [Ideas]
This will be a good challenge to create a powerful and scalable full-stack application using the Turbo monorepo setup! The project can maybe include a web application, a React Native Expo app, and a shared API extracted into the packages folder. Here are some of the  stack that can be leveraged to fast-forward the development:

Authentication: Clerk to handle user authentication seamlessly across both web and mobile. maybe also showcaing how to use jwt templates.
Database:  Drizzle ORM for structured type-safe queries with Supabase as the backend database and real-time provider.
API: tRPC to define and share type-safe API endpoints across the entire stack.

Idea is to highlight the benefits of a monorepo architecture, where shared logic and types live in the packages folder, enabling code reuse and faster development.

Features to implement could include:
User registration and login using Clerk.
A database schema with Drizzle and Supabase.
API endpoints for CRUD operations with tRPC.
Separate UI components for web and mobile, while sharing business logic.

**Comment by VadimNotJustDev** [comment]
This is an amazing idea. Thank you for the detailed breakdown of it, that helps
